server/src/main/java/org/apache/druid/catalog/model/CatalogUtils.java

```diff
@@ -68,12 +68,20 @@ public static Granularity asDruidGranularity(String value)
     if (Strings.isNullOrEmpty(value) || value.equalsIgnoreCase(DatasourceDefn.ALL_GRANULARITY)) {
       return Granularities.ALL;
```

Comment:
Hmm, why is no granularity treated as `ALL` granularity?

Reply:
Looking at the usages, it seems that `asDruidGranularity` is called only when the input granularity string is non-null (the argument is also _not_ marked `@Nullable`). If that's true, could you remove the `Strings.isNullOrEmpty(value)` conditional? Treating it as `ALL` when granularity isn't specified doesn't seem right...

Reply:
good thought. fixed